app/lib/utils.rb

```diff
@@ -54,30 +58,32 @@ def file_mime_type_by_name(entry_name:, is_form: false)
   end
 
   def file_extension_by_mime_type(mime_type)
-    case mime_type.downcase
-    when 'application/pdf'
+    return unless mime_type
+
+    case
+    when mime_type.include?('application/pdf')
       '.pdf'
-    when 'application/xml', 'application/x-eform-xml'
+    when %w[application/xml application/x-eform-xml application/vnd.gov.sk.xmldatacontainer+xml].any? { |xml_mimetype| mime_type.include?(xml_mimetype) }
       '.xml'
-    when 'application/vnd.etsi.asic-e+zip'
+    when mime_type.include?('application/vnd.etsi.asic-e+zip')
       '.asice'
-    when 'application/vnd.etsi.asic-s+zip'
+    when mime_type.include?('application/vnd.etsi.asic-s+zip')
       '.asics'
-    when 'application/x-xades_zep'
+    when mime_type.include?('application/x-xades_zep')
       '.xzep'
-    when 'application/x-zip-compressed'
+    when mime_type.include?('application/x-zip-compressed')
       '.zip'
-    when 'text/plain'
+    when mime_type.include?('text/plain')
```

Comment:
Tieto include mi pridu mierne nebezpecne ked tam bude napriklad nieco ako substring tak by to chytilo. Preco to nevieme matchovat priamo ako cely string?

Reply:
Ked tak mozeme include zmenit ta `start_with?`, ale ako cely string matchovat nechceme kvoli tomu, ze sa ukazalo, ze tie mimetypes byvaju aj v podobe napr.: `application/vnd.gov.sk.xmldatacontainer+xml; charset=UTF-8`. Ta cast `; charset=UTF-8` nam to kazi.

Reply:
Mozeme, pozriem